File: app/src/main/java/me/kirimin/mitsumine/network/RequestQueueSingleton.java
package me.kirimin.mitsumine.network;

import android.content.Context;

import com.android.volley.RequestQueue;
import com.android.volley.toolbox.Volley;

public class RequestQueueSingleton {

    private static RequestQueue queue;

    private RequestQueueSingleton() {
    }

    public static RequestQueue getRequestQueue(final Context context) {
        if (queue == null) {
            queue = Volley.newRequestQueue(context);
        }
        return queue;
    }
}

File: app/src/main/java/me/kirimin/mitsumine/network/RequestQueueSingleton.kt
package me.kirimin.mitsumine.network

import android.content.Context

import com.android.volley.RequestQueue
import com.android.volley.toolbox.Volley

public class RequestQueueSingleton private() {
    companion object {

        private var queue: RequestQueue? = null

        public fun get(context: Context): RequestQueue {
            if (queue == null) {
                queue = Volley.newRequestQueue(context)
            }
            return queue!!
        }
    }
}
